cilium/cmd/debuginfo.go

```diff
@@ -77,14 +96,82 @@ var sections = map[string]addSection{
 
 func init() {
 	rootCmd.AddCommand(debuginfoCmd)
-	debuginfoCmd.Flags().StringVarP(&file, "file", "f", "", "Redirect output to file")
-	debuginfoCmd.Flags().StringVarP(&html, "html-file", "", "", "Convert default output to HTML file")
+	debuginfoCmd.Flags().BoolVarP(&outputToFile, "file", "f", false, "Redirect output to file(s)")
 	debuginfoCmd.Flags().BoolVarP(&filePerCommand, "file-per-command", "", false, "Generate a single file per command")
+	debuginfoCmd.Flags().StringSliceVar(&outputOpts, "output", []string{}, "markdown| html| json| jsonpath='{}'")
+	debuginfoCmd.Flags().StringVar(&outputDir, "output-directory", "", "directory for files (if specified will use directory in which this command was ran)")
+}
+
+func validateInput() {
+	if outputDir != "" && !outputToFile {
+		fmt.Fprintf(os.Stderr, "invalid option combination; specified output-directory %q, but did not specify for output to be redirected to file; exiting\n", outputDir)
+		os.Exit(1)
+	}
+	validateOutputOpts()
+}
+
+func validateOutputOpts() {
+
+	for _, outputOpt := range outputOpts {
+		switch strings.ToUpper(outputOpt) {
+		case markdownOutput:
+		case htmlOutput:
+			if !outputToFile {
+				fmt.Fprintf(os.Stderr, "if HTML is specified as the output format, it is required that you provide the `--file` argument as well\n")
+				os.Exit(1)
+			}
+		case jsonOutputDebuginfo:
+			if filePerCommand {
+				fmt.Fprintf(os.Stderr, "%s does not support dumping a file per command; exiting\n", jsonOutputDebuginfo)
+				os.Exit(1)
+			}
+		case jsonpathOutput:
+			if filePerCommand {
+				fmt.Fprintf(os.Stderr, "%s does not support dumping a file per command; exiting\n", jsonpathOutput)
+				os.Exit(1)
+			}
+		default:
+			// Check to see if arg contains jsonpath filtering as well.
```

Comment:
This comment is helpful, thanks! Though the comment was dropped in the equivalent later usage. 

Along the lines of my above feedback on the naming of `re`, I think that this could be made more readable via something like:

```
if jsonPathRegex.MatchString(...) {
```